modules/javafx.graphics/src/main/java/javafx/animation/Timeline.java

```diff
@@ -191,11 +191,9 @@ void doStart(boolean forceSync) {
         clipCore.start(forceSync);
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
-    public void stop() {
+    void stopOnFxThread() {
+        // The parent check is redone in the super method. Consider refactoring.
```

Comment:
As you've probably already seen, this causes infinite recursion and an eventual StackOverflow. The easiest (and probably safest) fix is to revert this part of the change, keeping the overridden `stop` method, and have it call a private `stopOnFxThread` method (and make the one in the parent class also private). You can file a bug to refactor it later.

Reply:
I just pushed a fix that calls the correct super method. The implementation isn't fully tested yet because I wanted to get the CSR moving in parallel.

I'm still thinking on the least intrusive way to write the tests. I seems that the only way to check if a method runs on the FX thread is to override a method on its stack and check there. That means that I would need to expose (package-private) the `play/start/stop/pauseOnFxThread` methods.

Reply:
It works now with change to call the right super method.

As for the tests, I'd sooner not do anything intrusive in this PR, especially given the tight window to get this in. I'm not even sure whether it's worth inspecting the internals to make sure it is on the right thread in all cases, but if so, a follow-up testbug could be filed for that.